packages/plugin-ext/src/hosted/node/plugin-reader.ts

```diff
@@ -54,6 +59,18 @@ export class HostedPluginReader implements BackendApplicationContribution {
                     res.status(404).send(`No such file for plugin with id '${escape_html(pluginId)}'.`);
                 });
             } else {
+                // Requested resource is not local. Check providers if any.
```

Comment:
Strange that it is just a fallback of local logic, usually we use `URI` and decide based on schema for example, in this case based on authority would make sense, i.e. if `req.params.path` has some special scheme like `theia-remote` and authority which points to a remote host and port then it should be handled by a remote plugin service. Could be something like that done?